infrastructure/oss/types.go

```diff
@@ -243,6 +290,58 @@ func (i *ossIssue) AddSnykLearnAction(learnService learn.Service, ep error_repor
 	return action
 }
 
+func (i *ossIssue) AddQuickFixAction(affectedFilePath string, issueRange snyk.Range) *snyk.CodeAction {
+	if !config.CurrentConfig().IsSnyOSSQuickFixCodeActionsEnabled() {
+		return nil
+	}
+	log.Debug().Msg("create deferred quickfix code action")
+	quickfixEdit := i.getQuickfixEdit()
+	if quickfixEdit == "" {
+		return nil
+	}
+	upgradeMessage := "Upgrade to " + quickfixEdit + " (Snyk)"
+	autofixEditCallback := func() *snyk.WorkspaceEdit {
+		edit := &snyk.WorkspaceEdit{}
+		singleTextEdit := snyk.TextEdit{
+			Range:   issueRange,
+			NewText: quickfixEdit,
+		}
+		edit.Changes = make(map[string][]snyk.TextEdit)
+		edit.Changes[affectedFilePath] = []snyk.TextEdit{singleTextEdit}
+		return edit
+	}
+
+	action, err := snyk.NewDeferredCodeAction(upgradeMessage, &autofixEditCallback, nil)
+	if err != nil {
+		log.Error().Msg("failed to create deferred quickfix code action")
+		return nil
+	}
+	return &action
+}
+
+func (i *ossIssue) getQuickfixEdit() string {
```

Comment:
Haven't tested with maven and gradle projects yet, but I see potential problems with gradle build.gradle.kts files, as they are reported with package manager `maven`, IIRC.

Reply:
You're right, gradle in general was not being done properly in this piece of code. And also `build.gradle.kts` files were reported as `maven` instead of `gradle` so that's another edge case. I've got a fix ready but I'm wondering if we want to include the one for `build.gradle.kts`, since it's a symptom of https://snyksec.atlassian.net/browse/OSM-1775?

Reply:
There's two options: we can code around it by checking the extension `kts` or we can disable gradle automatic fixing for now. I think the linked issue is not gonna be resolved soon, so we should not wait.

Reply:
You're right. I've pushed a fix in https://github.com/snyk/snyk-ls/pull/511/commits/0968daed7f9df7b8f70f0e78bdef8836739f90cd

I tested using https://github.com/snyk/gradle-java-reach-goat too, which has a `build.gradle` file that also got interpreted as maven instead of gradle.
<img width="813" alt="Screenshot 2024-06-27 at 13 21 37" src="https://github.com/snyk/snyk-ls/assets/81559517/e3fbbe30-f475-4d8e-a9a8-821c7397a372">